File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    printf("Number: ");
    long credit_card = get_long_long();
    
    int count = 0;
    long long digits = credit_card;
    long long digits_two = credit_card;
    int sum = 0;
    int number_of_digits = 0;
    int first_digit;
    int check = 0;
    int i = 0;
    int j = 0;
    int k = 0;
    
    
    while(digits > 0)
    {
        digits = digits / 10;
        number_of_digits++;
    }
    
    while(digits_two > 9)
    {
        digits_two = digits_two / 10;
        first_digit = digits_two;
    }
    
    if (number_of_digits != 13 && number_of_digits != 15 && number_of_digits != 16)
    {
        printf("INVALID\n");
        check = 1;
    }
    
    while (credit_card > 0)
    {
        if(count % 2 == 0)
        {
            sum = sum + (credit_card % 10);
            credit_card = credit_card / 10;
            count++;
        }
        else
        {
            i = credit_card % 10 * 2;
            if (i > 9)
            {
                j = i % 10;
                k = i / 10;
                sum = sum + (j + k);
                credit_card = credit_card / 10;
                count++;
            }
            else
            {
            sum = sum + i;
            credit_card = credit_card / 10;
            count++;
            }
        }
    }
    
    if (sum % 10 == 0)
    {
        if(first_digit == 3)
        {
            printf("AMEX\n");
        }
        if(first_digit == 5)
        {
            printf("MASTERCARD\n");
        }
        if(first_digit == 4)
        {
            printf("VISA\n");
        }
    }
    else if (check == 0)
    {
        printf("INVALID\n");
    }
}

File: pset6/credit.py
import cs50

print("Number: ", end="")
credit_card = cs50.get_int()
credit_card_list = []
number_of_digits = 0
checksum = 0

#this get the digits and makes the nnumber into a list
while(credit_card != 0):
    number_of_digits += 1
    credit_card, r = divmod(credit_card, 10)
    credit_card_list.insert(0,r)

#checks if number of digits are valid
if(number_of_digits != 13 and number_of_digits != 15 and number_of_digits != 16):
    print("INVALID")
#run through digits and get checksum
else:
    for i in range(number_of_digits):
        if(number_of_digits % 2 == 0):
            if(i % 2 == 0):
                if(credit_card_list[i] > 4): #if product is more than two digits
                    checksum += credit_card_list[i] * 2 % 10 + 1
                else:
                    checksum += credit_card_list[i] * 2
            else:
                checksum += credit_card_list[i]
        else:
            if(i % 2 != 0):
                if(credit_card_list[i] > 4):
                    checksum += credit_card_list[i] * 2 % 10 + 1
                else:
                    checksum += credit_card_list[i] * 2
            else:
                checksum += credit_card_list[i]
    
    #see if checksum amounts to a real card
    if(checksum % 10 == 0):
        if(credit_card_list[0] == 3):
            print("AMEX");
        if(credit_card_list[0] == 5):
            print("MASTERCARD");
        if(credit_card_list[0] == 4):
            print("VISA");
    else:
        print("INVALID")